File: app/src/main/java/ru/mamykin/foreignbooksreader/ui/adapters/viewholders/FileViewHolder.java
package ru.mamykin.foreignbooksreader.ui.adapters.viewholders;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.ButterKnife;
import ru.mamykin.foreignbooksreader.R;

public class FileViewHolder extends RecyclerView.ViewHolder {
    @BindView(R.id.tvFileName)
    public TextView tvFileName;
    @BindView(R.id.tvFileAttributes)
    public TextView tvFileAttributes;
    @BindView(R.id.ivFileType)
    public ImageView ivFileType;
    @BindView(R.id.pbLoading)
    public View pbLoading;

    public FileViewHolder(View itemView, OnItemClickListener listener) {
        super(itemView);

        ButterKnife.bind(this, itemView);
        itemView.setOnClickListener(v -> listener.onItemClick(getAdapterPosition()));
    }

    public interface OnItemClickListener {
        void onItemClick(int position);
    }
}

File: app/src/main/java/ru/mamykin/foboreader/ui/global/FileViewHolder.kt
package ru.mamykin.foboreader.ui.global

import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.ImageView
import android.widget.TextView

import butterknife.BindView
import butterknife.ButterKnife
import ru.mamykin.foboreader.R

class FileViewHolder(itemView: View, listener: OnItemClickListener) : RecyclerView.ViewHolder(itemView) {
    @BindView(R.id.tvFileName)
    var tvFileName: TextView? = null
    @BindView(R.id.tvFileAttributes)
    var tvFileAttributes: TextView? = null
    @BindView(R.id.ivFileType)
    var ivFileType: ImageView? = null
    @BindView(R.id.pbLoading)
    var pbLoading: View? = null

    init {

        ButterKnife.bind(this, itemView)
        itemView.setOnClickListener { v -> listener.onItemClick(adapterPosition) }
    }

    interface OnItemClickListener {
        fun onItemClick(position: Int)
    }
}